pandas/core/indexes/category.py

```diff
@@ -414,7 +408,7 @@ def astype(self, dtype, copy=True):
             if dtype == self.dtype:
                 return self.copy() if copy else self
 
-        return super().astype(dtype=dtype, copy=copy)
```

Comment:
Is there a consistent reason to use Index instead of super?

Reply:
IIRC i got test failures without this change